cloud-init/write-archive-artifact-data-to-bucket.tpl.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+# This is a Terraform template file, and the input variables are
+# passed in via templatefile().
+#
+# shellcheck disable=SC2154
+
+# This script is used write out a bash script that can be used to create an
+# assessment artifact archive and copy it to the appropriate S3 bucket.
+
+# Input variables are:
+# * artifact_export_bucket_name - the name of the assessment artifact export S3
+#   bucket
+# * artifact_export_path - the path to copy the artifact to in the S3 bucket
+# * assessment_id - the identifier for the assessment
+# * permissions - the octal permissions to assign the script
+# * vnc_username - the username associated with the VNC user
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+path=/home/${vnc_username}/archive-artifact-data-to-bucket.sh
+
+# Write the script.  Note that we wrap the delimiter in quotes to
+# prevent shell variable substitution.
+cat > "$path" << "EOF"
+#!/usr/bin/env bash
+
+# This script creates a gzipped tar archive of the directory containing
+# assessment artifacts and then copies that archive to the appropriate S3
+# bucket.
+#
+# Usage: archive-artifact-data-to-bucket.sh /path/to/artifacts_directory
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ $# -ne 1 ]; then
+  echo "Usage: archive-artifact-data-to-bucket.sh /path/to/artifacts_directory"
+  exit 1
+fi
+
+full_bucket_path="s3://${artifact_export_bucket_name}/${artifact_export_path}-${assessment_id}.tgz"
+
+# Prompt for confirmation
+read -p "Confirm: Archive the contents of $1 and upload to $full_bucket_path? [y/N] " -n 1 -r
+echo  # Move to a new line
+
+if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+  echo "Exiting without archiving or copying data to S3."
+  exit 1
+fi
+
+# Change to the parent directory of the artifacts directory
+cd "$(dirname "$1")"
+
+# Create the archive
+echo "Creating archive..."
+# Since we are in the parent directory of the artifacts directory, we don't
+# expect to run into any issues related to disk space (in our typical use case,
+# this directory resides on the EFS volume, which should not run out of space).
+# If disk space does become a problem for some reason, this script can be
+# modified to create the archive in a temporary directory.
```

Comment:
If the EFS volume can't contain the archive, then certainly `/tmp` can't either.
```suggestion
# Since we are in the parent directory of the artifacts directory, we don't
# expect to run into any issues related to disk space.  (In our typical use 
# case, this directory resides on the EFS volume, which is huge.)
```

Reply:
You also make a good point, now that you have explained it.  How about this?
```suggestion
# Since we are in the parent directory of the artifacts directory, we don't
# expect to run into any issues related to disk space.  (In our typical use 
# case, this directory resides on the EFS volume, which is huge.)
#
# The important thing is to avoid filling up the root disk, so in the event
# that an EFS volume _is not_ being used it probably makes sense to create 
# the archive in /tmp.
```